numpy/lib/arraysetops.py

```diff
@@ -551,6 +555,60 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     ar1 = np.asarray(ar1).ravel()
     ar2 = np.asarray(ar2).ravel()
 
+    # Convert booleans to uint8 so we can use the fast integer algorithm
+    if ar1.dtype == np.bool_:
+        ar1 = ar1.view(np.uint8)
+    if ar2.dtype == np.bool_:
+        ar2 = ar2.view(np.uint8)
+
+    # Check if we can use a fast integer algorithm:
+    integer_arrays = (np.issubdtype(ar1.dtype, np.integer) and
+                      np.issubdtype(ar2.dtype, np.integer))
+
+    if integer_arrays and _slow_integer in [None, False]:
+        ar2_min = np.min(ar2)
+        ar2_max = np.max(ar2)
+        ar2_size = ar2.size
+
+        # Check for integer overflow
+        with np.errstate(over='raise'):
+            try:
+                ar2_range = ar2_max - ar2_min
+
+                # Optimal performance is for approximately
+                # log10(size) > (log10(range) - 2.27) / 0.927, see discussion on
+                # https://github.com/numpy/numpy/pull/12065
+                optimal_parameters = (
+                        np.log10(ar2_size + 1.0) >
+                        ((np.log10(ar2_range + 1.0) - 2.27) / 0.927)
```

Comment:
Why the +1?

Reply:
For `ar2_range` of 0 it is a way to prevent blow-up. Then I add a 1 to the other log10 to balance out the original fit. But it doesn't change the "optimal region" that much.

Reply:
+1 makes complete sense for the 'ar2_range', since you use '+1' on it below when using it too. Would be nice to eliminate the '+1' on the size, since it doesn't have any reasonable interpretation.

Reply:
Okay sounds good.